### ISSUE TITLE
Unicode matching

### ISSUE THREAD

**Issue by mikaelho** [SonOfLilit/kleenexp#30]
Hello, and thanks for a very interesting package!

I would suggest that the `#letter` macro be tweaked so that it matches modern Python behaviour, where `\w` in a unicode pattern string matches a unicode letter like `ä`, whereas the current implementation `#letter` does not.

**Comment by SonOfLilit** [comment]
I agree with this suggestion and intend to do this in the future, it's a nontrivial task and I want to launch first.

**Comment by mikaelho** [comment]
Makes sense.

Your use of the word ”launch” and the stated goals of displacing re made me want to ask if you saw this [list](https://github.com/mikaelho/python-human-regex), also known as ”the competition”?

**Comment by SonOfLilit** [comment]
I saw it now, very well researched.

I noticed that in the "trying to use" section, some packages try to parse "hello (a-123)", some try to parse  "hello [a-123]", and kleenexp tries to parse  "hello [a-123)". How about using the expression:

```
[[capture:title 1+ #any] ' ' #tag=[[capture:key #letters] '-' [capture:id #digits]] ['(' #tag ')' | '[' #tag ']']]
```

To only parse those you actually want?

I maintain a list, in the readme (https://github.com/SonOfLilit/kleenexp#similar-works), of packages I consider competition. The serious ones that I hope might actually win are `melody` and `pomsky`. You'll notice all of your list is lumped under "There are many more eDSLs, but I will not list them as they are less relevant in my opinion"

I honestly just don't believe you can solve the problem I'm trying to solve for just a single language, you have to solve it for editor find&replace, for log monitoring system configuration, for database queries, etc'.

What would need to be done before kleenexp is the sole winner of your research? Wondering if my dev roadmap lines up with your needs :)

**Comment by SonOfLilit** [comment]
I saw it now, very well researched.

I noticed that in the "trying to use" section, some packages try to parse
"hello (a-123)", some try to parse  "hello [a-123]", and kleenexp tries to
parse  "hello [a-123)". How about using the expression:

```
[[capture:title 1+ #any] ' ' #tag=[[capture:key #letters] '-' [capture:id
#digits]] ['(' #tag ')' | '[' #tag ']']]
```

To only parse those you actually want?

I maintain a list, in the readme (
https://github.com/SonOfLilit/kleenexp#similar-works), of packages I
consider competition. The serious ones that I hope might actually win are
melody and pomsky. You'll notice all of your list is lumped under

   -

   There are many more eDSLs, but I will not list them as they are less
   relevant in my opinion

I honestly just don't believe you can solve the problem I'm trying to solve
for just a single language, you have to solve it for editor find&replace,
for log monitoring system configuration, for database queries, etc'.

What would need to be done before kleenexp is the sole winner of your
research? Wondering ifs my dev roadmap lines up with your needs :)
